### PR TITLE
Implement file watcher for new file service

### PR DESCRIPTION
**Pull request by bpasero**

### CODE REVIEW

**Comment by nathansobo** [comment]
Hi @bpasero. We're having stability issues with `nsfw` in Atom and took a look at what you guys are doing. It looks like you're now using `nsfw` by default for multi-root projects. Is there somewhere where I can learn about the context of that decision? We'd like to move *away* from `nsfw` and were curious about your .Net helper + Chokidar solution, but it seems like there's some shortcoming with that approach for multi-folder workspaces? Thanks.

**Comment by bpasero** [comment]
@nathansobo I would say our current state of file watching in VSCode is a bit crazy for historic reasons:
* back when we did not have multi root support, we would use [`chokidar`](https://github.com/microsoft/chokidar) on macOS/Linux and a custom implemented C# file watcher for windows [`vscode-filewatcher-windows`](https://github.com/microsoft/vscode-filewatcher-windows)
* starting with multi root, I thought it would be a good idea to try out a file watcher that was solid on all platforms to eventually replace our existing solutions and so I thought [`nsfw`](https://github.com/microsoft/vscode-nsfw) was a good idea

**Turns out:** none of our choices were ideal. `nsfw` has issues, and so does `chokidar`. The latter lately really got active again with the recent release of [version 3.0](https://github.com/paulmillr/chokidar/releases/tag/3.0.0) which claims to have much improved performance, which is one of our biggest remaining issues with chokidar to date (https://github.com/microsoft/vscode/issues/3998)

**Bottom line:** I want us to go back to a single library for file watching and I want to give `chokidar` a chance to be that one library if the recent performance improvements hold true. 

**PS:** for a long time I was really hoping https://github.com/atom/watcher would be that one library that could replace them all. What happened there?

**Comment by nathansobo** [comment]
My first reaction to the instability of `nsfw` was to try to default to `@atom/watcher`, but I also experienced stability issues on that branch. The author of that library is no longer on the Atom team and it isn't really used anywhere else. I decided that even though `nsfw` has stability issues, at least it has a proven track record and is actively maintained. I even tried to wrap the Rust `notify` crate in hopes that it would be a more solid solution, but alas it had problems as well and that was a huge waste. So we're back to `nsfw` with another band-aid, but I look at all the threads it's spawning in our test suite and wonder what the heck is going on.

It sounds like chokidar might be worth another look if we have further problems. There's nothing fundamental then around using it to watch multi-root projects? Does it support Windows now or will you continue to lean on the .Net executable?

**Comment by bpasero** [comment]
Yeah `chokidar` does support Windows and in the end it boils down to node.js watch facility (which itself is leveraging the same `readdirectorychangesw` as our implementation does, so I do not expect much of a difference). Only on macOS (as far as I know) is `chokidar` using `fsevents` directly. 

I would like to end up with just one file watcher if possible.
